peachjam/admin.py

```diff
@@ -437,7 +437,22 @@ def get_queryset(self, request):
     def get_fieldsets(self, request, obj=None):
         fieldsets = super().get_fieldsets(request, obj)
         if obj is None:
-            fieldsets = self.new_document_form_mixin.adjust_fieldsets(fieldsets)
+            if request.user.has_perm("peachjam.can_edit_advanced_fields"):
```

Comment:
I think we always need to call `adjust_fieldsets`. The manipulation of the fieldsets based on the user permission should happen whether or not `obj` is None.

So these two things seem unrelated to me, meaning you can move your changes to below the if block, and always do it.